scripts/test_hardware/test_integration.py

```
@@ -0,0 +1,247 @@
+        raise ValueError(f"Unsafe URL scheme: {parsed.scheme!r}; only http/https are allowed")
+    req = urllib.request.Request(url)
+    try:
+        with urllib.request.urlopen(req, timeout=REQUEST_TIMEOUT) as resp:
+            return resp.status, resp.read().decode("utf-8", errors="replace")
+    except urllib.error.HTTPError as e:
+        return e.code, e.read().decode("utf-8", errors="replace")
+
+
+def _post_json(url: str, data: dict) -> tuple[int, str]:
+    body = json.dumps(data).encode("utf-8")
+    req = urllib.request.Request(url, data=body, method="POST")
```

Ruff (0.15.2)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)